tools/walletextension/rpcapi/filter_api.go

```diff
@@ -186,32 +189,80 @@ func (api *FilterAPI) NewFilter(crit common.FilterCriteria) (rpc.ID, error) {
 }
 
 func (api *FilterAPI) GetLogs(ctx context.Context, crit common.FilterCriteria) ([]*types.Log, error) {
-	logs, err := ExecAuthRPC[[]*types.Log](
-		ctx,
-		api.we,
-		&ExecCfg{
-			cacheCfg: &CacheCfg{
-				CacheTypeDynamic: func() CacheStrategy {
-					// when the toBlock is not specified, the request is open-ended
-					if crit.ToBlock != nil && crit.ToBlock.Int64() > 0 {
-						return LongLiving
-					}
-					return LatestBatch
-				},
-			},
-			tryUntilAuthorised: true,
-			adjustArgs: func(acct *GWAccount) []any {
-				// convert to something serializable
-				return []any{common.FromCriteria(crit)}
+	method := "eth_getLogs"
```

Comment:
I ended up not using the ExecAuthRPC function for this, because the logic is different from anything else